src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -20,11 +24,148 @@ public InstructorCourseDetailsPageData(AccountAttributes account) {
 
     public InstructorAttributes currentInstructor;
     public CourseDetailsBundle courseDetails;
-    public List<StudentAttributes> students;
     public List<InstructorAttributes> instructors;
     public String studentListHtmlTableAsString;
+    public ElementTag giveCommentButton;
+    public ElementTag courseRemindButton;
+    public CourseDetailsStudentsTable studentsTable;
     
-    public String getInstructorCourseRemindLink() {
+    public void init(InstructorAttributes currentInstructor, CourseDetailsBundle courseDetails, 
+                     List<InstructorAttributes> instructors, List<StudentAttributes> students) {
+        this.currentInstructor = currentInstructor;
+        this.courseDetails = courseDetails;
+        this.instructors = instructors;
+        
+        boolean isDisabled = !currentInstructor.isAllowedForPrivilege(
+                                                    Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        
+        String content = "<span class=\"glyphicon glyphicon-comment glyphicon-primary\"></span>";
+        giveCommentButton = createButton(content, "btn btn-default btn-xs icon-button pull-right", 
+                                         "button_add_comment", null, "", "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        String onClick = "if(toggleSendRegistrationKeysConfirmation('" 
+                          + Sanitizer.sanitizeForJs(courseDetails.course.id) + "')) "
+                          + "window.location.href='" 
+                          + Sanitizer.sanitizeForJs(getInstructorCourseRemindLink()) + "';";
+        courseRemindButton = createButton(null, "btn btn-primary", "button_remind", null, 
+                                          Const.Tooltips.COURSE_REMIND, "tooltip", onClick, isDisabled);
+        
+        studentsTable = new CourseDetailsStudentsTable();
+        int studentIndex = 0;
+        for (StudentAttributes student : students) {
+            CourseDetailsStudentsTableRow row = createStudentsTableRow(studentIndex, student);
+            studentsTable.getRows().add(row);
+            studentIndex++;
+        }
+    }
+    
+    public CourseDetailsStudentsTableRow createStudentsTableRow(int studentIndex, 
+                                                                StudentAttributes student) {
+        CourseDetailsStudentsTableRow row = new CourseDetailsStudentsTableRow();
+        
+        row.setStudent(student);
```

Comment:
Would it be easier to put the `student` String as a variable in the constructor for `CourseDetailsStudentsTableRow`?